### ISSUE TITLE
Toast is showing vertically in Landscape iPad

### ISSUE THREAD

**Issue by SSamanta** [devxoul/JLToast#79]
![error](https://cloud.githubusercontent.com/assets/3451021/15170844/4a7cca76-1767-11e6-987e-7f57000b6b97.png)

```
JLToastView.setDefaultValue(
            UIColor.redColor(),
            forAttributeName: JLToastViewBackgroundColorAttributeName,
            userInterfaceIdiom: .Pad
        )
        let toast = JLToast.makeText("Testing Toast long toast long toast long toast..")
        toast.show()
```

**Comment by devxoul** [comment]
@SSamanta, thanks for report! I'll take a look soon. Which version of iPad are you using?

**Comment by SSamanta** [comment]
iPad Air

**Comment by devxoul** [comment]
And iOS version please :)

**Comment by SSamanta** [comment]
Latest ios 9.3.1. My apps orientation mode only landscape left or right

**Comment by 24601** [comment]
+1. This is happening on iPad Pro (12.1") as well as in Simulator. 

Seems to have happened from the last time I updated JLToast.

**Comment by SSamanta** [comment]
yes. Before it was working correctly.

**Comment by devxoul** [comment]
@SSamanta, are you not using LaunchScreen?

<img width="498" alt="screen shot 2016-05-14 at 3 52 09 am" src="https://cloud.githubusercontent.com/assets/931655/15258568/41200336-1987-11e6-8104-7ec80c265184.png">

**Comment by devxoul** [comment]
@SSamanta, Could you please try with `ipad-orientation-launchscreen` branch? If you're using CocoaPods, you can just edit your **`Podfile`**:

```ruby
use_frameworks!

target 'YourProject' do

pod 'JLToast', :git => 'https://github.com/devxoul/JLToast', :branch => 'ipad-orientation-launchscreen'

end
```

and install

```bash
$ pod install
```

**Comment by 24601** [comment]
@devxoul - we use a LaunchScreen and I updated to the ipad-orientation-launchscreen branch and the issue still exists.

**Comment by devxoul** [comment]
@24601, could you please provide me a sample project that I can test with?

**Comment by SSamanta** [comment]
@devxoul I tried LaunchScreen and pod 'JLToast', :git => 'https://github.com/devxoul/JLToast', :branch => 'ipad-orientation-launchscreen'
end both..same issue it's showing in vertically..I am trying to make one sample app..

**Comment by willyjl** [comment]
This bug also happens on my code and the cause is due to `statusBarOrientation` is deprecated, based on https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIApplication_Class/#//apple_ref/c/tdef/UIInterfaceOrientation

Whenever the following code is run in JLToastView on landscape orientation and iOS 9.3:

```
        let orientation = UIApplication.sharedApplication().statusBarOrientation
        if orientation.isPortrait || !JLToastWindow.sharedWindow.shouldRotateManually {
            width = containerSize.width
            height = containerSize.height
            y = portraitOffsetY
        } else {
            width = containerSize.height
            height = containerSize.width
            y = landscapeOffsetY
        }
```

The conditional always goes to portrait, never landscape.

**Comment by willyjl** [comment]
I have found the root causes. If you select all device orientations and tap Off `Requires full screen`, it works fine. But if you either deselect any of device orientations or tap On `Requires full screen`, this bug happens.

**Comment by SSamanta** [comment]
Ohoo my requirement is landscape and full screen so second case . Are there any way to fix that?

**Comment by devxoul** [comment]
@SSamanta @24601 @willyjl Could you guys please replace `shouldRotateManually` property of the `JLToastWindow` with the code above and try again?

```swift
/// Don't rotate manually if the application:
///
/// - is running on iOS 9
/// - supports all orientations
/// - doesn't require full screen
///
var shouldRotateManually: Bool {
    let application = UIApplication.sharedApplication()
    let window = application.delegate?.window ?? nil
    let orientationMask = application.supportedInterfaceOrientationsForWindow(window)
    let requiresFullScreen = NSBundle.mainBundle().infoDictionary?["UIRequiresFullScreen"]?.boolValue == true
    if #available(iOS 9, *), orientationMask == .All && !requiresFullScreen {
        return false
    }
    return true
}
```

I hope it works 🙏

**Comment by willyjl** [comment]
Yup, I tried your solution and it works. Good job @devxoul! Thanks for the quick response. 👍

**Comment by devxoul** [comment]
@SSamanta @24601 @willyjl I just released [1.4.1](https://github.com/devxoul/JLToast/releases/tag/1.4.1)! If it has problem, please reopen this issue :smile:
